I don't know how to solve this:

> Test 'runnable\testmodule.d' failed. The logged output:
D:\a\1\s\generated\windows\release\64\dmd.exe -conf= -m64 -Irunnable   -odD:\a\1\s\test\test_results\runnable -ofD:\a\1\s\test\test_results\runnable\testmodule_0.exe  runnable\testmodule.d 
>
>
>   Creating library D:\a\1\s\test\test_results\runnable\testmodule_0.lib and object D:\a\1\s\test\test_results\runnable\testmodule_0.exp
>
>
>
> testmodule_0.exp : error LNK2001: unresolved external symbol _D3run17unicode_06_å“ªé‡Œ12__ModuleInfoZ
>
>
>
>  Hint on symbols that are defined and could potentially match:
>
>
>
>    _D3run17unicode_06_哪里12__ModuleInfoZ
>
>
>
> D:\a\1\s\test\test_results\runnable\testmodule_0.exe : fatal error LNK1120: 1 unresolved externals

--- a/test/dshell/test9377.d
+++ b/test/dshell/test9377.d
@@ -1,8 +1,8 @@
 import dshell;
 void main()
 {
-    Vars.set("libname", "$OUTPUT_BASE/a$LIBEXT");
+    Vars.set("libname", "$OUTPUT_BASE/a_b$LIBEXT");
 
     run("$DMD -m$MODEL -I$EXTRA_FILES -of$libname -c $EXTRA_FILES/mul9377a.d $EXTRA_FILES/mul9377b.d -lib");
-    run("$DMD -m$MODEL -I$EXTRA_FILES -of$OUTPUT_BASE/a$EXE $EXTRA_FILES/multi9377.d $libname");
+    run("$DMD -m$MODEL -I$EXTRA_FILES -of$OUTPUT_BASE/a_b_full$EXE $EXTRA_FILES/multi9377.d $libname");
 }

--- a/test/runnable/gdb15729.sh
+++ b/test/runnable/gdb15729.sh
@@ -7,8 +7,12 @@ echo RESULT=
 p s.val
 "
 
-$DMD -g -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${LIBEXT} -lib ${EXTRA_FILES}${SEP}lib15729.d
-$DMD -g -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}${SEP}gdb15729.d ${OUTPUT_BASE}${LIBEXT}
+
+libname=${OUTPUT_BASE}_dep${LIBEXT}
+
+
+$DMD -g -m${MODEL} -I${EXTRA_FILES} -of${libname} -lib ${EXTRA_FILES}${SEP}lib15729.d
+$DMD -g -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}${SEP}gdb15729.d ${libname}
 
 if [ $OS == "linux" ]; then
     echo "${GDB_SCRIPT}" > ${OUTPUT_BASE}.gdb

--- a/test/runnable/link14198a.sh
+++ b/test/runnable/link14198a.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 
-libname=${OUTPUT_BASE}${LIBEXT}
+libname=${OUTPUT_BASE}_dep${LIBEXT}
 
 
 # build library

--- a/test/runnable/link14834.sh
+++ b/test/runnable/link14834.sh
@@ -3,7 +3,7 @@
 
 dir=${RESULTS_DIR}${SEP}runnable
 
-libname=${OUTPUT_BASE}${LIBEXT}
+libname=${OUTPUT_BASE}_dep${LIBEXT}
 exename=${OUTPUT_BASE}${EXE}
 
 $DMD -m${MODEL} -I${EXTRA_FILES} -lib           -of${libname} ${EXTRA_FILES}${SEP}link14834a.d

--- a/test/runnable/link846.sh
+++ b/test/runnable/link846.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 
-libname=${OUTPUT_BASE}${LIBEXT}
+libname=${OUTPUT_BASE}_dep${LIBEXT}
 
 # build library with -release
 $DMD -m${MODEL} -I${EXTRA_FILES} -of${libname} -release -boundscheck=off -lib ${EXTRA_FILES}${SEP}lib846.d

--- a/test/runnable/test10386.sh
+++ b/test/runnable/test10386.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 
-libname=${OUTPUT_BASE}${LIBEXT}
+libname=${OUTPUT_BASE}_dep${LIBEXT}
 
 
 $DMD -m${MODEL} -Irunnable -I${EXTRA_FILES} -of${libname} -c ${EXTRA_FILES}${SEP}lib10386${SEP}foo${SEP}bar.d ${EXTRA_FILES}${SEP}lib10386${SEP}foo${SEP}package.d -lib

--- a/test/runnable/test13666.sh
+++ b/test/runnable/test13666.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 
-libname=${OUTPUT_BASE}${LIBEXT}
+libname=${OUTPUT_BASE}_dep${LIBEXT}
 
 
 $DMD -m${MODEL} -I${EXTRA_FILES} -of${libname} -lib ${EXTRA_FILES}${SEP}lib13666.d

--- a/test/runnable/test13742.sh
+++ b/test/runnable/test13742.sh
@@ -1,8 +1,11 @@
 #!/usr/bin/env bash
 
 
-$DMD -m${MODEL} -I${EXTRA_FILES} -lib -cov -of${OUTPUT_BASE}${LIBEXT} ${EXTRA_FILES}${SEP}lib13742a.d ${EXTRA_FILES}${SEP}lib13742b.d
-$DMD -m${MODEL} -I${EXTRA_FILES} -cov -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}${SEP}test13742.d ${OUTPUT_BASE}${LIBEXT}
+libname=${OUTPUT_BASE}_dep${LIBEXT}
+
+
+$DMD -m${MODEL} -I${EXTRA_FILES} -lib -cov -of${libname} ${EXTRA_FILES}${SEP}lib13742a.d ${EXTRA_FILES}${SEP}lib13742b.d
+$DMD -m${MODEL} -I${EXTRA_FILES} -cov -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}${SEP}test13742.d ${libname}
 
 ${OUTPUT_BASE}${EXE} --DRT-covopt=dstpath:${RESULTS_TEST_DIR}
 

--- a/test/runnable/test18456.sh
+++ b/test/runnable/test18456.sh
@@ -1,8 +1,12 @@
 #!/usr/bin/env bash
 
+
+libname=${OUTPUT_BASE}_dep${LIBEXT}
+
+
 # source file order is important
-$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${LIBEXT} -lib ${EXTRA_FILES}/lib18456b.d ${EXTRA_FILES}/lib18456.d
-$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}/test18456.d ${OUTPUT_BASE}${LIBEXT}
+$DMD -m${MODEL} -I${EXTRA_FILES} -of${libname} -lib ${EXTRA_FILES}/lib18456b.d ${EXTRA_FILES}/lib18456.d
+$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}/test18456.d ${libname}
 ${OUTPUT_BASE}${EXE}
 
 rm_retry ${OUTPUT_BASE}{${LIBEXT},${EXE}}

--- a/test/runnable/test21723.sh
+++ b/test/runnable/test21723.sh
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 
-$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${LIBEXT} -lib ${EXTRA_FILES}/lib21723a.d ${EXTRA_FILES}/lib21723b.d
-$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} -inline ${EXTRA_FILES}/test21723.d ${OUTPUT_BASE}${LIBEXT}
+
+libname=${OUTPUT_BASE}_dep${LIBEXT}
+
+
+$DMD -m${MODEL} -I${EXTRA_FILES} -of${libname} -lib ${EXTRA_FILES}/lib21723a.d ${EXTRA_FILES}/lib21723b.d
+$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} -inline ${EXTRA_FILES}/test21723.d ${libname}
 
 rm_retry ${OUTPUT_BASE}{${LIBEXT},${EXE}}

--- a/test/runnable/test23148.sh
+++ b/test/runnable/test23148.sh
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 
-$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${LIBEXT} -lib ${EXTRA_FILES}/lib23148.d
-$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}/test23148.d ${OUTPUT_BASE}${LIBEXT}
+
+libname=${OUTPUT_BASE}_dep${LIBEXT}
+
+
+$DMD -m${MODEL} -I${EXTRA_FILES} -of${libname} -lib ${EXTRA_FILES}/lib23148.d
+$DMD -m${MODEL} -I${EXTRA_FILES} -of${OUTPUT_BASE}${EXE} ${EXTRA_FILES}/test23148.d ${libname}
 
 rm_retry ${OUTPUT_BASE}{${LIBEXT},${EXE}}

--- a/test/runnable/testmodule.d
+++ b/test/runnable/testmodule.d
@@ -1,5 +1,10 @@
 // PERMUTE_ARGS:
 
+// MS linker apparently doesn't (properly?) support Unicode in
+//      `/INCLUDE:symbol` linker directives.
+//      This has been upstreamed from LDC.
+// DISABLED: win
+
 // $HeadURL$
 // $Date$
 // $Author$

--- a/test/tools/d_do_test.d
+++ b/test/tools/d_do_test.d
@@ -1730,13 +1730,22 @@ int tryMain(string[] args)
                 }
             }
 
+            /// This should be similar in nature as ``dshell_prebuilt.d`` ``filterCompilerOutput`` function.
             void prepare(ref string compile_output)
             {
                 if (compile_output.empty)
                     return;
 
                 compile_output = compile_output.unifyNewLine();
-                compile_output = std.regex.replaceAll(compile_output, regex(`^DMD v2\.[0-9]+.*\n? DEBUG$`, "m"), "");
+                compile_output = std.regex.replaceAll(compile_output, std.regex.regex(`^DMD v2\.[0-9]+.*\n? DEBUG$`, "m"), "");
+
+                if(envData.os == "windows" && envData.model != "32omf") {
+                    // For MSVC linker, when it generates a binary file with an export table (with the import generation) it will add a message
+                    // this message is both linker specific and platform specific, which we do not want our tests checked against.
+                    // so we'll remove it prior to doing anything with the output.
+                    compile_output = std.regex.replaceAll(compile_output, std.regex.regex(r"\s*Creating library [\S\\/]+ and object [\S\\/]+\r?\n?", "s"), "");
+                }
+
                 compile_output = std.string.strip(compile_output);
                 // replace test_result path with fixed ones
                 compile_output = compile_output.replace(result_path, resultsDirReplacement);

--- a/test/tools/dshell_prebuilt/dshell_prebuilt.d
+++ b/test/tools/dshell_prebuilt/dshell_prebuilt.d
@@ -326,10 +326,20 @@ private GrepResult grepLines(T)(T lineRange, string finalPattern)
 
 /**
 remove \r and the compiler debug header from the given string.
+
+This should be the same code that is used in ``d_do_test.d`` inside the inline function ``tryMain`` ``testCombination`` ``prepare``.
 */
 string filterCompilerOutput(string output)
 {
     output = std.string.replace(output, "\r", "");
-    output = std.regex.replaceAll(output, regex(`^DMD v2\.[0-9]+.*\n? DEBUG\n`, "m"), "");
+    output = std.regex.replaceAll(output, std.regex.regex(`^DMD v2\.[0-9]+.*\n? DEBUG\n`, "m"), "");
+
+    version(Windows) {
+        // For MSVC linker, when it generates a binary file with an export table (with the import generation) it will add a message
+        // this message is both linker specific and platform specific, which we do not want our tests checked against.
+        // so we'll remove it prior to doing anything with the output.
+        output = std.regex.replaceAll(output, std.regex.regex(r"\s*Creating library [\S\\/]+ and object [\S\\/]+\r?\n?", "s"), "");
+    }
+
     return output;
 }

